CLDR-15608 Specify ranges of numbers with qualifiers (#2381)
unicode-org/cldr: docs/ldml/tr35-numbers.md

Two semantically equivalent tokens can be *collapsed* if they appear at the start of both values or the end of both values. However, the implementation may choose different levels of aggressiveness with regard to collapsing tokens. The currently recommended heuristic is:

1. Never collapse scientific or compact notation. This is to avoid producing ambiguous strings such as "3–5M" (could represent 3–5,000,000 or 3,000,000–5,000,000).
1. Only collapse if the tokens are more than one code point in length. This is to increase clarity of strings such as "$3–$5".

To perform the collapse:

1. Remove the token that is closest to the range separator.
That is, for a prefix element, remove from the end value, and for a suffix element remove it from the start value:
    * USD 2 – USD 5 ⇒ USD 2 – 5

>     * 2M EUR – 5M EUR ⇒  2 M – 5M EUR

Reviewer: ```suggestion
    * 2M EUR – 5M EUR ⇒  2M – 5M EUR
```

Author: done

    * 2 km – 5 km ⇒ 2 – 5 km
    * 2M ft – 5M ft ⇒ 2M – 5M ft

> 2. In bidi contexts, rule #1 is applied **visually**. For example, if a range from 2 km to 5 km would be presented visually as "_mk 5 – mk 2_", the collapsed form would be "_mk 5 – 2_". (The _mk_ is a stand-in for the native representation.)

Reviewer: Question about the new bidi clause. The example you gave should be fine for both visual and logical order, right? In the ICU implementation we definitely only do this in logical order (because visual order would require pulling in the bidi code, which we don't do).

Author: Revised, with the bidi note restructured.

> 2. In bidi contexts, rule #1 is applied **visually**. For example, if a range from 2 km to 5 km would be presented visually as "_mk 5 – mk 2_", the collapsed form would be "_mk 5 – 2_". (The _mk_ is a stand-in for the native representation.)

Reviewer 2: ```suggestion
2. In bidi contexts, rule #1 is applied logically. For example, if a range from 2 km to 5 km would be presented visually as "_mk 5 – mk 2_",  `(٢كم-٥كم)` the collapsed form would be "_mk 5 – 2_"  `(٢-٥كم)`. (The _mk_ is a stand-in for the native representation.)
```
my suggestion rendered below:

<img width="935" alt="image" src="https://user-images.githubusercontent.com/855219/192581342-f0e8e070-24fd-408b-bee8-c3c4c418514a.png">

> The removal of  a token requires modification of the remaining token for the correct plural form, if the token can have distinct plural forms. That is, use [Plural Ranges](#Plural_Ranges) to calculate the correct form for the remaining token, and pick the variant of that token corresponding to that plural form.

Reviewer: ```suggestion
3. If necessary, modify the remaining token with the correct plural form, if the token can have distinct plural forms. That is, use [Plural Ranges](#Plural_Ranges) to calculate the correct form for the remaining token, and pick the variant of that token corresponding to that plural form.
```

Author: Reworded as suggested.
